src/query/ts/m3db/consolidators/step_accumulator.go

```diff
@@ -35,7 +35,7 @@ type StepLookbackAccumulator struct {
 	lookbackDuration time.Duration
 	stepSize         time.Duration
 	earliestLookback time.Time
-	datapoints       [][]xts.Datapoint
+	datapoints       []xts.Datapoint
```

Comment:
Nice, this was able to be flattened?

Reply:
Yeah, I was initially trying to abstract away the `i-th` wise stuff in the generic step iterator and let this accumulator handle it, but that doesn't lend very well to parallelization